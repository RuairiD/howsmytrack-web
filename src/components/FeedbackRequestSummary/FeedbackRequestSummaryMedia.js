import React from "react";

import { Alert } from "antd";
import MediaEmbed from "../MediaEmbed/MediaEmbed";

const FeedbackRequestSummaryMedia = ({ mediaUrl, mediaType }) => {
    if (!mediaUrl) {
        return (
            <Alert
                message="No track has been provided for this request. You'll be able to join a group and provide feedback for others, but you won't receive any feedback for yourself."
                type="info"
                showIcon
            />
        );
    }
    if (!mediaType) {
        return (
            <Alert
                message="Unable to play media. Invalid URL."
                type="error"
                showIcon
            />
        );
    }
    return <MediaEmbed mediaUrl={mediaUrl} mediaType={mediaType} />;
};

export default FeedbackRequestSummaryMedia;
