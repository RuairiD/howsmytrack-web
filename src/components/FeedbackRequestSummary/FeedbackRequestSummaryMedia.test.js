import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackRequestSummaryMedia from "./FeedbackRequestSummaryMedia";

describe("FeedbackRequestSummaryMedia", () => {
    it("shows an information message if mediaUrl is missing/request is trackless", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryMedia />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("shows an error if mediaType is missing (implies the url is malformed or not supported)", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryMedia
                mediaUrl="mediaUrl"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a MediaEmbed for the given mediaUrl and mediaType", () => {
        const wrapper = shallow(
            <FeedbackRequestSummaryMedia
                mediaUrl="mediaUrl"
                mediaType="mediaType"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
