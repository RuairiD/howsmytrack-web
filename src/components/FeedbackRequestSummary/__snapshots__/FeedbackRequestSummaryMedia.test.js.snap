// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackRequestSummaryMedia renders a MediaEmbed for the given mediaUrl and mediaType 1`] = `
<MediaEmbed
  mediaType="mediaType"
  mediaUrl="mediaUrl"
/>
`;

exports[`FeedbackRequestSummaryMedia shows an error if mediaType is missing (implies the url is malformed or not supported) 1`] = `
<Alert
  message="Unable to play media. Invalid URL."
  showIcon={true}
  type="error"
/>
`;

exports[`FeedbackRequestSummaryMedia shows an information message if mediaUrl is missing/request is trackless 1`] = `
<Alert
  message="No track has been provided for this request. You'll be able to join a group and provide feedback for others, but you won't receive any feedback for yourself."
  showIcon={true}
  type="info"
/>
`;
